qemu/qemu.go

```diff
@@ -127,6 +152,41 @@ func appendQMPSocket(params []string, config Config) []string {
 	return params
 }
 
+func appendDevices(params []string, config Config) []string {
+	for _, d := range config.Devices {
+		if d.Type != "" {
+			var deviceParams []string
+
+			deviceParams = append(deviceParams, fmt.Sprintf("%s", d.Type))
+
+			if d.ID != "" {
```

Comment:
With a stronger typing, rather than:

if d.ID != "" {
}

it could be something like:

if d.ID.IsValid() {
}

where the validator function knows how to check for a zero value, as well as any other appropriate formatting and range correctness.